haskell.nix

```diff
@@ -0,0 +1,110 @@
+# A flake-parts module for Haskell cabal projects.
+{ config, lib, flake-parts-lib, ... }:
+
+let
+  inherit (flake-parts-lib)
+    mkSubmoduleOptions
+    mkPerSystemOption;
+  inherit (lib)
+    mkOption
+    mkDefault
+    types;
+in
+{
+  options = {
+    perSystem = mkPerSystemOption
+      ({ config, self', inputs', pkgs, system, ... }: {
+        options.haskellProject = {
+          haskellPackages = mkOption {
+            type = types.anything;
+            description = ''Which Haskell package set to use'';
+            default = pkgs.haskellPackages;
+          };
+          name = mkOption {
+            type = types.str;
+            description = ''Name of the Cabal package'';
+          };
+          root = mkOption {
+            type = types.path;
+            description = ''Path to the Cabal project root'';
+            default = ./.;
+          };
+          overrides = mkOption {
+            type = types.anything;
+            description = ''Overrides for the Cabal project'';
+            default = self: super: { };
+          };
+          modifier = mkOption {
+            type = types.anything;
+            description = ''Modifier for the Cabal project'';
+            default = drv: drv;
+          };
+          baseBuildTools = mkOption {
+            type = types.anything;
+            description = ''Common build tools for Haskell develop; you don't need to set this.'';
+            default = hp: with hp; [
+              cabal-install
+              haskell-language-server
+              ghcid
+              hlint
+              cabal-fmt
+            ];
+          };
+          extraBuildTools = mkOption {
+            type = types.anything;
+            description = ''Extra tools to add to nix-shell'';
+            default = hp: [ ];
+          };
```

Comment:
You could unify these into a single option `buildTools` of type `attrsOf (nullOr package)`. This will make it easier for users to override the defaults, e.g. `buildTools.ghcid = null;`. Use `attrValues` on the option value where you use it.